lua/neogit/buffers/status/ui.lua

```diff
@@ -5,6 +5,7 @@ local common = require("neogit.buffers.common")
 local config = require("neogit.config")
 local a = require("plenary.async")
 local state = require("neogit.lib.state")
+local git = require("neogit.lib.git")
```

Comment:
I'd like to keep all data-fetching out of the view layer. I know some views break this rule, currently, but the eventual goal is to have the views be purely presenters, with all state (repo data to present) fetched up-front and passed into the constructor. 

The compromise for this, if the data is _very_ expensive to fetch, is to set it up to lazy-load via metatable `__index` method. There are a few examples of this scattered around. That keeps knowledge of fetching data out of the view layer, even if it's triggered there.